### DISCUSSION TITLE
Upgrading ArgoCD via helm

### DISCUSSION THREAD

**Discussion by soakes** [argoproj/argo-helm#2759] [Q&A]
Hi everyone,

I'm encountering an error while trying to upgrade ArgoCD from version 7.0.0 to the latest version using Helm. I'm hoping someone can help me troubleshoot the issue.

Error Message:
```
❯ helm upgrade --reuse-values argocd argo/argo-cd
Error: UPGRADE FAILED: template: argo-cd/templates/redis/service.yaml:20:6: executing "argo-cd/templates/redis/service.yaml" at <include "argo-cd.dualStack" .>: error calling include: template: argo-cd/templates/_helpers.tpl:253:16: executing "argo-cd.dualStack" at <.Values.global.dualStack.ipFamilyPolicy>: nil pointer evaluating interface {}.ipFamilyPolicy
```

Additional Information:
- Current ArgoCD chart Version: 7.0.0
- Upgrade Target: Latest Version

Has anyone else encountered this error during an ArgoCD upgrade? Any insights or solutions would be greatly appreciated!

Thanks,

Simon

**Comment by jwilkicki** [comment]
I got around something like that by using --reset-then-reuse-values  rather than --reuse-values

**Reply by soakes** [reply]
Thank you for the reply.  
I did in the end find a work around which was:

```
helm get values <release-name> -n argocd > current-values.yaml
helm upgrade <release-name> argo/argo-cd --namespace argocd --version <chart-version> -f current-values.yaml
```

If I recall correctly.  However was trying to reuse it without dumping the current values.
